src/main/java/bio/terra/service/snapshot/flight/delete/DeleteSnapshotPopAndLockDatasetStep.java

```diff
@@ -78,11 +78,9 @@ public StepResult doStep(FlightContext context) {
 
   @Override
   public StepResult undoStep(FlightContext context) {
+    FlightMap workingMap = context.getWorkingMap();
     try {
-      UUID sourceDatasetId =
-          snapshotService.getFirstSourceDatasetIdFromSnapshotId(
-              snapshotId, authenticatedUserRequest);
-
+      UUID sourceDatasetId = workingMap.get(DatasetWorkingMapKeys.DATASET_ID, UUID.class);
```

Comment:
This field is written in this step's `doStep()`. If, for some reason, the step fails after the lock occurs but before the FlightMap is written, can this `get()` return `null`?

Reply:
We only lock the dataset after we write the id to the working map, so if there's no dataset id, then it doesn't need unlocking.

Reply:
@ddietterich this is the case I was wondering about in our chat before. From what you told me, this dependency of the `undoStep()` on the `doStep()` is fine and will work correctly, is that right?